### ISSUE TITLE
Reorganize components and remove validation on constants

### ISSUE THREAD

**Issue by dinghino** [dinghino/stocks-historical-data#44]
In order to actually allow custom components to be added we need to remove the
validation performed on the `scraper.settings.constants`.

Those can be kept for internal reference, for now at least, but it would be
better to rethink on how that works, maybe even reorganizing the code.

The validation itself is technically not needed anymore, I think, since every
component passes through the manager to be handled, so the names can actually be
anything now.

It would also be a good idea to actually define naming and import schemes to follow.

Also, to improve code navigation and import we could change the packages structure
to something like

```bash
├── components/
│   │   # should import, like it does now, `fetchers` and `writers`,
│   │   # at least for now and include `constants`
│   ├── __init__.py
│   ├── component_base.py
│   ├── base_fetcher.py
│   ├── base_parser.py
│   ├── base_parser.py
│   │
│   │   # one folder for each source to handle
│   ├── <source_name>
│   │   │   # imports classes and constants, especially the source_name
│   │   ├── __init__.py
│   │   ├── <source_name>_fetcher.py
│   │   ├── <source_name>_parser.py
│   │   │   # local constants and a SOURCE_NAME for matching
│   │   └── constants.py
│   │
│   │   # one folder for each output type/writer
│   ├── <output_type>/
│   │   ├── __init__.py
│   │   ├──<writer_name>.py
│   │   │   # local constants and OUTPUT_TYPE for matching
│   │   └──constants.py
```

This could allow anybody to create a new set of handlers for a source/output type,
like for example

```bash
├── <my_source>/
│   ├── __init__.py
│   ├── <my_source>_fetcher.py
│   ├── <my_source>_parser.py
│   └── constants.py
```
```python
# classes creation
from scraper.components import Fetcher  # base class

class MyFetcher(Fetcher):
  pass
```

```python
from scraper.components import manager
from my_source import MyParser, MyFetcher, SOURCE_NAME

# Add them to the manager so that both the app and the cli can use them
manager.register_handler(SOURCE_NAME, MyFetcher, MyParser)
```

since the interface will be the same regardless and matching is now done through the manager, working this way it's simpler to set up new set of components and organize them.

**Comment by dinghino** [comment]
Solved with PR #50 . Still need to reorganize tests in submodules for components but that's the lesser part

**Comment by dinghino** [comment]
validation through constants is still present. reopening

**Comment by dinghino** [comment]
Will probably remove all the lists and related validations from settings since they can be easily handled from the manager, interfacing the settings directly with manager for validation on set methods.

for example, the `add_source` checks on `settings.sources` which is technically redundant. the list of sources can be retrieved from the manager, like the cli does, so it has not much sense duplicating the value.

One though could be to include the manage as a property of settings and/or app too (and even components) so that it can be accessed directly from whenever those components are used

**Comment by dinghino** [comment]
finally closed with #54 

There are still things to do to improve, especially cli side, but for now it works with what we have
